scripts/py_matter_yamlclusters/matter_yamlclusters/cluster/delay_commands.py

```diff
@@ -0,0 +1,33 @@
+#
```

Comment:
What is the rationale for having this be a separate module?
`scripts/py_matter_yamlclusters/matter_yamlclusters/cluster/`?

Why not just scripts/py_matter_yamltests/matter_yamltests/cluster/

Reply:
Just to be a little more verbose. I get that some people might want to decided not to use the defaults. But the defaults could be provided with the yamltests, and if someone wants to create their own they can do so outside of this distributable python packages in general

Reply:
@tehampson The rationale is that some consumers of the `matter_yamltests` package  may want to have different implementation for those clusters depending on their test harness setup or the limitation for testing on real devices.

I tried to explained that shortly in the issue description (but I did failed apparently):
```
The custom implementation lives into its own package since different consumer of matter_yamltests may do different things (e.g there is no mechanism to reboot a real world device directly, while it can be done in CI, hence the different implementations that may exists for CI vs certification).
```

We need to have some ways for a consumer to override the default clusters if needed - that said If you feel strongly about that I don't mind having a *default* implementation that lives into `matter_yamltests` ?

Reply:
Right now I don't see the value in creating the default somewhere else unless there is a good reason like it pulls in other python package that some people might not want to pull in. Because even if it is in the yamltests package, people can still choose not to use it and create their own wherever makes sense for them

Maybe lets see if anyone else has an opinion

Reply:
@raju-apple @krypton36 Do you have an opinion on the packaging ?


Basically this is about pseudo-clusters (e.g `LogCommands`,  `SystemCommands`, etc...). I was assuming that your needs does not necessarily fits the CI needs and that it would be nice for you if you can override them with a custom package.
I did have built a package for the "default" implementation as an example and that may be overkill - just want to know what is the more practical thing for you.

Reply:
I believe that we can separate a BUILD.gn rule if several other BUILD.gn rulese will make use of it (like idl parsing is used by parser and tests).

In this particular case, the only user of pseudo clusters seems to be yaml tests and a separate implementation of these seems still to be `use yaml tests with a separate implementation`.

I would lean for this reason towards single gn package - all usages of these seem related to yamltests.  I am not quite sure how a separate wheel simplifies things here.

Reply:
I would totally see that separate module though - these are independent. I believe a separate module fits the "we can override" while not needing a separate BUILD.gn/wheel.

Reply:
I would prefer to have them in separate modules, my reasoning is that later down the line if we would like to add plugins/adapters/overrides for the test harness for certain functionality in cluster actions, it would make things much easier and more obvious what the TH needs vs what the framework is using (default).

Reply:
@krypton36 Will TH use this as a standalone module outside of using yamltests? If so in what way were you thinking

Reply:
Also, this will allow us to be able to create an adapter cluster that can be used by vendors to reboot their own accessory by their own means. We use server based commands to Linux to reboot, but other might use toggle power on power strip, etc...

Reply:
>I am thinking for adding ability to send "chip-tool" commands/read/write outside of the YAML test.

Will this tool that is "outside of YAML tests" need these pseudo clusters? If so then maybe renaming this module should actually be called `pseudoclusters` instead `matterclusters`. Additionally if should not be dependent on `python_deps = [ "${chip_root}/scripts/py_matter_yamltests:matter_yamltests" ]`

If not, what does that tool you are asking for have to do with these pseudo cluster? Whever you want there can be done outside `yamlclusters`. 

Just so this is said explicitly right now yamltest should not be tightly coupled to chip-tool for sending thing to the accessory. Chip-tool should be replaceable by any other tool that is able to construct whatever needs to be sent out on the wire to communicate with the accessory.

>this will allow us to be able to create an adapter cluster that can be used by vendors to reboot their own accessory by their own means.

Having this in a separate package doesn't prevent anyone from creating their own pseudo cluster for their own use outside of yamltests. Not following what you are implying here, can you please clarify some more?

Reply:
^ @krypton36 (not sure if you monitor these discussion only when mentioned, sorry for the spam if this was already on your radar but were going to reply later)

Reply:
I spoke to @vivien-apple offline and we discussed that for TH use cases, it is agnostic to this being in separate package or in the current package.

Reply:
I made it a module.